### PR TITLE
Add createapplication with arbitrary params

### PR DESCRIPTION
**Pull request by eguzki**
### What
The existing `CreateApp` method did not allow adding extra parameters. Some parameters that are allowed at the creation time are not allowed at update time. For instance: `application_id`.

### Verification Steps

Run the following code to create a new application

```go
func main() {
        c := helper.Generic_client()      //     client.NewThreeScale(adminPortal, accessToken)   

	var (
		accountID   int64  = ****
		planID      int64  = ****
		name        string = "myApp"
		description string = "myApp"
	)

	params := client.Params{
		"application_id": "something:special",
		"description":    description,
	}

	obj, err := c.CreateApplication(accountID, planID, name, params)
	if err != nil {
		panic(err)
	}

}
```

### CODE REVIEW

File: client/app.go [hunk 1]
```diff
@@ -26,17 +26,26 @@ const (
...
+// params:
+// * description: Description of the application to be created.
+// * user_key: User Key (API Key) of the application to be created.
+// * application_id: App ID or Client ID (for OAuth and OpenID Connect authentication modes) of the application to be created.
+// * application_key: App Key or Client Secret (for OAuth and OpenID Connect authentication modes) of the application to be created.
+// * redirect_url: Redirect URL for the OAuth request.
+// * first_traffic_at: Timestamp of the first call made by the application.
+// * first_daily_traffic_at: Timestamp of the first call on the last day when traffic was registered for the application ('Traffic On').
+func (c *ThreeScaleClient) CreateApplication(accountId, planId int64, name string, params Params) (Application, error) {
 	var app Application
```
Comment (tkan145) [inline client/app.go:39]: Wouldn't it is better to bring `apiResp := &ApplicationElem{}` up here and return `apiResp.Application` instead? Nothing major though, just a style reference

File: client/app.go [hunk 2]
```diff
@@ -26,17 +26,26 @@ const (
...
+// CreateApplication - Create an application.
+// params:
+// * description: Description of the application to be created.
+// * user_key: User Key (API Key) of the application to be created.
+// * application_id: App ID or Client ID (for OAuth and OpenID Connect authentication modes) of the application to be created.
+// * application_key: App Key or Client Secret (for OAuth and OpenID Connect authentication modes) of the application to be created.
+// * redirect_url: Redirect URL for the OAuth request.
+// * first_traffic_at: Timestamp of the first call made by the application.
+// * first_daily_traffic_at: Timestamp of the first call on the last day when traffic was registered for the application ('Traffic On').
+func (c *ThreeScaleClient) CreateApplication(accountId, planId int64, name string, params Params) (Application, error) {
```
[the remark this answers is not in the document]
Comment (eguzki) [reply client/app.go:38]: That's a good question. 

3scale IDs are big numbers. Specially when you read to SaaS instance IDs. For me they should all be strings, but some response from 3scale API are integers (json integers and not strings). For 32 bit arch's, int32 is not big enough. As you know `int` size is platform dependent, so int32 would be too small to accommodate 3scale IDs. For 64bits, `int` should be good.

File: client/app.go (hunk 1 above)
Comment (eguzki) [reply client/app.go:39]: That's functionally equivalent, nothing changes.  `var app Application` is just "empty" application, used when returning error.

**Comment by MStokluska** [comment]
Verified it on cluster and it works as expected. 
/lgtm

**Review by MStokluska — APPROVED** [review]

**Pull request merged**